Project/Source/Resources/ResourceStateMachine.cpp

```diff
@@ -189,3 +202,11 @@ Transition* ResourceStateMachine::FindTransitionGivenName(const std::string& nam
 
 	return nullptr;
 }
+
+ResourceStateMachine::~ResourceStateMachine() {
+
+	if (initialState) {
+		delete (initialState);
+		initialState = nullptr;
+	}
```

Comment:
- The list of states is leaking, make sure you release them too.
- This should be done in the Unload function.
- You can use the RELEASE macro to do the nullptr check and assignment for you.

Reply:
The states list is a list of objects not a list of pointers 	std::list<State> states; should it be also cleared?